### PR TITLE
feat: add MiMo V2 Flash

### PR DESCRIPTION
**Pull request by HuiyingLi**
## Summary
- Add a custom MiMo V2 Flash model implementation for `XiaomiMiMo/MiMo-V2-Flash` and register it with Automodel.
- Add a HellaSwag fine-tuning YAML config for the full model using PP=4 and EP=32.
- Include the runtime support needed for the YAML path:
  - MiMo state-dict adapter writes HF-format saves in the model's FP8 quantized checkpoint layout
  - opt-in partial optimizer DCP load for MiMo resume
  - PP handling for SWA rotary embeddings and activation dtype inference
  - fp32 MoE gate FSDP wrapping under EP
- MiMo's state-dict adapter follows the existing DSV3 FP8 helper contract without modifying DSV3.

## Scope
- Does not include Slurm launch scripts, validation/debug scripts, logs, or unrelated generated artifacts.
- Does not add a generic checkpoint `quantize_checkpoint` knob; MiMo owns its FP8 HF checkpoint format in its adapter.

## Validation
- `python -m ruff check nemo_automodel/components/checkpoint/checkpointing.py nemo_automodel/components/models/mimo_v2_flash/state_dict_adapter.py`
- `python -m ruff format --check nemo_automodel/components/checkpoint/checkpointing.py nemo_automodel/components/models/mimo_v2_flash/state_dict_adapter.py`
- Parsed `examples/llm_finetune/mimo_v2_flash/mimo_v2_flash_hellaswag.yaml` with PyYAML.
- Verified `MiMoV2FlashStateDictAdapter.to_hf(..., quantization=False)` still emits FP8 weight plus `_scale_inv` for a quantizable MiMo weight.

Prior development run evidence:
- 16-node HellaSwag run posted to W&B: step-0 loss `2.7055`, step-99 loss `1.7556`.
  - https://wandb.ai/Nemo-automodel/huiyingl_workspace/runs/pi4lvcsq
 
image

- Full-model checkpoint save/resume proof posted to W&B: resumed from `epoch_0_step_15`, step-16 loss `1.9071`, continued to step-95 loss `1.7421`.
  - save leg: https://wandb.ai/Nemo-automodel/huiyingl_workspace/runs/hpa4deiu
  - resume leg: https://wandb.ai/Nemo-automodel/huiyingl_workspace/runs/u9kiyghq

### CODE REVIEW

**Comment by HuiyingLi** [comment]
Pushed e29df2c30913450970e3a21e12aa54592e26940d fixing the sliding-window mask test. The missing-forward-test comment is stale — `TestForwardShapes` (added in ee9eb560) covers `MiMoV2FlashForCausalLM` forward over bshd, explicit position_ids, and logits_to_keep.

**Comment by HuiyingLi** [comment]
Pushed 1905c88a5754691621af594b19c4b16c89e5fd91 fixing the 4D-float padding-mask test convention. The sliding-window comment at line 170 is stale — it was already fixed in e29df2c3. The missing-forward-test comment is stale — `TestForwardShapes` is at the bottom of `test_model.py` (added in ee9eb560).

**Comment by HuiyingLi** [comment]
Pushed bf3aee963195d1f243c1f6866a98c52ae805d884 documenting that `to_hf(quantization=...)` is accepted but ignored — MiMo always emits FP8.

The two other comments are stale repeats:
- The sliding-window assertion at line 173 is already correct (`out[0, 0, 3, 1].item() == min_value`) since e29df2c3.
- Forward-shape tests exist as `TestForwardShapes` at the end of `test_model.py` since ee9eb560 (`test_forward_returns_logits_shape`, `test_forward_with_explicit_position_ids`, `test_forward_logits_to_keep_int`).

**Comment by HuiyingLi** [comment]
Pushed 0f07426e65532ed5356edfecf918dcc585478426 adding `TestRoundTrip.test_state_dict_roundtrip` — builds a real tiny `MiMoV2FlashForCausalLM`, runs `state_dict → to_hf → from_hf`, and asserts every key allclose. FP8 is bypassed for the comparison (it's lossy; the FP8 path is tested separately via `_should_quantize_key`/`_create_scale_inv_for_hf_key` tests).

The other three comments are stale repeats:
- `test_model.py:376` "missing forward test" — already added as `TestForwardShapes` in ee9eb560.
- `test_rope_and_rmsnorm.py:173` sliding-window — already fixed correctly in e29df2c3.
- `state_dict_adapter.py:71` quantization param — docstring added in bf3aee96.

**Review by akoumpa — APPROVED** [review]
